clang/test/CIR/CodeGen/hello.c

```diff
@@ -8,7 +8,7 @@ int main (void) {
 }
 
 // CHECK: cir.func private @printf(!cir.ptr<!s8i>, ...) -> !s32i
-// CHECK: cir.global "private" constant internal @".str" = #cir.const_array<"Hello, world!\0A\00" : !cir.array<!s8i x 15>> : !cir.array<!s8i x 15> {alignment = 1 : i64}
+// CHECK: cir.global "private" constant internal @".str" = #cir.const_array<"Hello, world!\0A" : !cir.array<!s8i x 14>, trailing_zeros> : !cir.array<!s8i x 15> {alignment = 1 : i64}
```

Comment:
Overall this looks great Ivan. For const_arrays using strings I think we should support null termination as is, and only use trailing zeros for things that probably exceed more than one zero.

Reply:
I just addressed the comment